docs/DeveloperGuide.md

```diff
@@ -4,11 +4,212 @@
+
+```TaskList``` stores items that belong to 4 subclasses that draw upon a shared superclass.
+
+```Task```: The skeleton superclass of all Tasks. 
+
+-   ```ToDo```: Basic form of ```Task```
+
+-   ```Deadline```: Expands upon ```Task``` by storing a ```by``` variable
+
+-   ```Event```: Expands upon ```Task``` by storing ```start``` and ```end``` variable
+
+-   ```SchoolClass```: Expands upon ```Task``` by storing ```start``` and ```end``` variable, along with a ```className``` variable
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+This section describes how Duck is implemented and how its commands are executed.
+
+### Purge feature
+
+**Sequence Diagram**
+
+![image](https://user-images.githubusercontent.com/1620654/227128864-cfc9cac7-bc68-4962-ac45-77e1ae067db0.png)
+
+The ```TaskList#purge()``` command will be implemented following the sequence diagram as shown above.
+
+
+**Implementation**
+
+The ```TaskList#purge()``` command will be implemented to facilitate the removal of expired ```Task``` items from the ```TaskList```. This will be executed once upon Duck's startup, and can be further executed by inputting the term ```purge``` into the CLI when Duck is running. The following is the new operation to be implemented.
+
+-   ```TaskList#purge()``` - Navigates through the ```TaskList``` array, and checks the date/time of each Task item within. If the date/time of a given ```Task``` has passed, it will be removed from the ```TaskList``` array. The user will be then able to view a total of how many ```Tasks``` were purged, along with what ```Task``` they were.
+
+<br />
+
+**Given below is an example usage scenario for ```purge()```**
+
+**Step 1.** The user launches the application for the first time. The ```TaskList``` will be initialized with the data from a given pre-existing datafile if it exists, and the ```Task``` items will be inserted into the ```TaskList```.![](https://lh5.googleusercontent.com/XcnArNZc5ZqHhxXZ1iRuq0aRxNKkcD_snVgDv-dlpMVYFQFQjyGXdtHWvFGzG-k2UW5SXkjQbthIRtkrol_SYsNgmsmKw1kKEkME0vSQ0tfRfUIWj3jZTTVrerHbESuA-AlTpPEkj8JrKl-mMlu-3yM)
+
+**Step 2.** The ```TaskList#purge()``` command will be executed. The process is illustrated through two steps:
+
+**Identifying all expired ```Task``` items in ```TaskList```.**
+
+![](https://lh3.googleusercontent.com/Kbz7j3EmMGJGrIX1aQ8DCkHG8Y3gaIx3Ohf-zmYvr-S-LrxRqt8P1wlTMWUnlnna2tnR0i72yUJWB72Z2mWwGuwXiD_RnlIpYEk-MY6k9zMaLqTk_VkyHaRwIuzxi1trXHX1ySdZWBqZFDaCZ6K-QrY)
+
+**Removing all expired ```Task``` items in ```TaskList```.**
+![](https://lh4.googleusercontent.com/kVOcvIKAF1nlLSHXQsRPDuG6RwiyaqRXf9CYTLrv2WUezegCNx05GIC1KPYME6Eojid1hRbqUIzHUjKWRpYjUG7zCYE8586kLzbIrVdjZVnC5j_1ke1WBdFZSHoar_MuXKBA9eMARxCOgcja5qjP2_A)
+
+The user will see this on his terminal after the ```purge``` command has executed.
+
+```
+----------------------------------------------------------------------
```

Comment:
is expired count part of TaskLIst or what? I am very confused as to what expiredCount is part of.